contracts/common/ScaleReputation.sol

```diff
@@ -0,0 +1,54 @@
+/*
+  This file is part of The Colony Network.
+
+  The Colony Network is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  The Colony Network is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with The Colony Network. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+pragma solidity 0.8.20; // ignore-swc-103
+import "../../lib/dappsys/math.sol";
+
+contract ScaleReputation is DSMath {
```

Comment:
Does this really need to be its own contract?

Reply:
Where else would you suggest it goes? In general, this should probably be more common than it currently is in the codebase. Longer term, I want to do a pass-through to make sure there isn't unused code being deployed with contracts due to too-generous inclusion of rarely used code (in terms of number of contracts that inherit the code vs use it).

Reply:
I agree that an audit is overdue. I would say though that a separate contract should contain at least 5 related functions. I would be in favor of pulling out other reputation-related functionality into `ColonyNetworkReputatation`

Reply:
It's shared between Colony and ColonyNetwork, though.

Reply:
Ok fair. Hopefully we can place more functionality here and rename it something more general in the future, `CommonUtils` or something along those lines.

Reply:
I'm not convinced that's a good solution; we'd then (presumably) be including code that isn't necessary in contracts? (i.e. a contract that only used this function would still have the code for all the others included).

Reply:
Hmm. Sounds like you're angling for more of a "mixin" approach, where we can pull a lot of small contracts in as needed. I guess that approach is defensible from a contract size perspective. I might want to see another directory, `common/mixins/` or something like that so they don't end up cluttering the directory structure.